app/demo-interactor/KNDemoIO.cc

```diff
@@ -0,0 +1,63 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file KNDemoIO.cc
+//---------------------------------------------------------------------------//
+#include "KNDemoIO.hh"
+
+#include <nlohmann/json.hpp>
+
+namespace demo_interactor
+{
+//---------------------------------------------------------------------------//
+//@{
+//! I/O routines for JSON
+void to_json(nlohmann::json& j, const CudaGridParams& v)
+{
+    j = nlohmann::json{{"block_size", v.block_size},
+                       {"grid_size", v.grid_size}};
+}
+
+void from_json(const nlohmann::json& j, CudaGridParams& v)
+{
+    j.at("block_size").get_to(v.block_size);
+    j.at("grid_size").get_to(v.grid_size);
+}
+
+void to_json(nlohmann::json& j, const KNDemoRunArgs& v)
+{
+    j = nlohmann::json{{"energy", v.energy},
```

Comment:
"fun" fact :) ... ROOT can produce JSON files without the need for those streaming functions.